### PR TITLE
Rename Module to F3, pass network name to Client, fix error wrapping

### PR DESCRIPTION
**Pull request by Kubuxu**
It simplifies the structure of F3, gpbftHost/Runner. gpbftHost becomes a newtype of gpbftRunner to isolate exposure of APIs.

### CODE REVIEW

**Comment by Kubuxu** [comment]
I know it is an ugly PR (with multiple changes) let me know if I should split of these oneliners.

File: gpbft/participant.go [hunk 1]
```diff
@@ -100,7 +102,7 @@ func (p *Participant) ValidateMessage(msg *GMessage) (valid ValidatedMessage, er
 
 	// Validate the message.
 	if err = ValidateMessage(comt.power, comt.beacon, p.host, msg); err != nil {
-		return nil, fmt.Errorf("%w: %w", ErrValidationInvalid, err)
+		return nil, xerrors.Errorf("%v: %w", err, ErrValidationInvalid)
```
Comment (masih) [inline gpbft/participant.go:105]: Can you please remind me what is the advantage of using `xerrors` over `fmt` in this case?

File: host.go [hunk 2]
```diff
@@ -30,14 +32,9 @@ type gpbftRunner struct {
...
 	log        Logger
-
-	host *gpbftHost
 }
 
-type gpbftHost struct {
-	Client
-	runner *gpbftRunner
-}
+type gpbftHost gpbftRunner
```
Comment (masih) [inline host.go:37]: Why the type alias?

File: host.go [hunk 3]
```diff
@@ -19,6 +19,8 @@ type Client interface {
 	Logger() Logger
 }
 
+// gpbftRunner is responsible for running gpbft.Participant, taking in all concurrent events and
+// passing them to gpbft in a single thread.
 type gpbftRunner struct {
```
Comment (masih) [inline host.go:24]: Would be nice to reduce concept here: gpbftHost/gpbftRunner/Module.

**Comment by masih** [comment]
Also please make sure PR description is in place as this repo uses squash merge and pics the PR description as commit message by default 🙏

File: gpbft/participant.go (hunk 1 above)
Comment (Kubuxu) [reply gpbft/participant.go:105]: It keeps track of line information, making debugging easier.
The key thing here is that the first `%w` was invalid.

File: host.go (hunk 2 above)
Comment (Kubuxu) [reply host.go:37]: To separate internal (towards gpbft.Participant) and external (towards f3.F3) APIs

File: host.go (hunk 3 above)
Comment (Kubuxu) [reply host.go:24]: Module got renamed F3, host and runner are the same thing just with different APIs exposed

Comment (masih) [reply host.go:24]: > Module got renamed F3

Has this change been pushed in this PR?

Comment (Kubuxu) [reply host.go:24]: Uhhh, I accidentally un-did it.

**Review by masih — APPROVED** [review]

File: cmd/f3/run.go [hunk 4]
```diff
@@ -98,3 +98,15 @@ func setupDiscovery(h host.Host) (closer func(), err error) {
 	s := mdns.NewMdnsService(h, DiscoveryTag, &discoveryNotifee{h: h})
 	return func() { s.Close() }, s.Start()
 }
+
+type fakeSigner struct {
```
[the remark this answers is not in the document]
Comment (Kubuxu) [reply cmd/f3/run.go:102]: I need to override the Marshal function.

**Pull request merged**